Add page metadata (#29051)
hashicorp/terraform: website/intro/index.html.markdown

If you are already familiar with the basics of Terraform, the
[documentation](/docs/index.html) provides a better reference
guide for all available features as well as internals.

> **Hands-on:** Try the [Get Started](https://learn.hashicorp.com/tutorials/terraform/infrastructure-as-code) tutorials on HashiCorp Learn.

## What is Terraform?

Terraform is a tool for building, changing, and versioning infrastructure
safely and efficiently. Terraform can manage existing and popular service
providers as well as custom in-house solutions.

Configuration files describe to Terraform the components needed to
run a single application or your entire datacenter.
Terraform generates an execution plan describing
what it will do to reach the desired state, and then executes it to build the
described infrastructure. As the configuration changes, Terraform is able
to determine what changed and create incremental execution plans which

> can be applied.

Reviewer: I think we lost the bit about Terraform comparing the infrastructure to the configuration to make changes. Maybe we can add that bit to the change automation section?

Author: Good call - thank you for flagging. I added that into that section - let me know what you think.

> Infrastructure is described using a high-level [configuration language](/docs/language/index.html) in human-readable, declarative configuration files. This allows you to create a blueprint that can be versioned, shared, and reused.

Reviewer: Active vice 

- is described
- can be versioned ...

Author: Thank you! fixed in my recent commit!

> Terraform generates an _execution plan_ describing what it will do and asks for your approval before creating, updating, or destroying infrastructure. This allows you to review changes before they are applied.

Reviewer: Active voice (are applied)

Author: Also fixed!

> Terraform builds a resource graph and parallelizes the creation and modification of any non-dependent resources. This allows Terraform to

Reviewer: Maybe "creates or modifies non-dependent resources in parallel"?

Author: yes! Changed that in my commit.

> build resources as efficiently as possible and gives operators greater insight into their infrastructure.

Reviewer: ```suggestion
build resources as efficiently as possible and gives you greater insight into your infrastructure.
```

Author: Done! I wasn't sure about this one because I went back and forth about whether the audience might be broader somehow because this is an overview page...but then I realized that it's likely operators who are this deep in reference docs. Thank you!

page_title: "Introduction"
sidebar_current: "what"
description: |-
  Learn what Terraform is, what problems it can solve, and how it compares to existing software.
---

# Introduction to Terraform

> ## What is Terraform?

Author: Yeah that's fair - I went back and forth on this - I'll remove it so it's shorter!

Welcome to the intro guide to Terraform! This guide is the best
place to start with Terraform. We cover what Terraform is, what
problems it can solve, how it compares to existing software,
and contains a quick start for using Terraform.

If you are already familiar with the basics of Terraform, the
[documentation](/docs/index.html) provides a better reference
guide for all available features as well as internals.

> **Hands-on:** Try the [Get Started](https://learn.hashicorp.com/tutorials/terraform/infrastructure-as-code) tutorials on HashiCorp Learn.

## What is Terraform?

Terraform is a tool for building, changing, and versioning infrastructure
safely and efficiently. Terraform can manage existing and popular service

> providers as well as custom in-house solutions.

Author: Got it - added that back in. See what you think!
